pkg/dependencies/dependents.go

```diff
@@ -0,0 +1,244 @@
+			pkgName := helpers.AllPkgTreeToPurl(&dependency.Package.AllPkgTree)
+
+			var depPkgIds []string
+			pkgId := dependency.Package.Namespaces[0].Names[0].Versions[0].Id
+
+			if len(dependency.DependencyPackage.Namespaces[0].Names[0].Versions) == 0 {
+				findMatchingDepPkgVersionIDs, err := FindDepPkgVersionIDs(ctx, gqlClient, dependency.DependencyPackage.Type,
+					dependency.DependencyPackage.Namespaces[0].Namespace,
+					dependency.DependencyPackage.Namespaces[0].Names[0].Name, dependency.VersionRange)
+				if err != nil {
+					return nil, fmt.Errorf("error from FindMatchingDepPkgVersionIDs:%w", err)
+				}
+				depPkgIds = append(depPkgIds, findMatchingDepPkgVersionIDs...)
+			} else {
+				depPkgIds = append(depPkgIds, dependency.DependencyPackage.Namespaces[0].Names[0].Versions[0].Id)
+			}
+
+			for _, depPkgId := range depPkgIds {
+				// Inside the loop where you iterate through dependencies
+				updatePackagesAndNames(idToName, packages, depPkgId, pkgId, depPkgName, pkgName, dependency.DependencyPackage.Type, dependency.DependencyPackage.Namespaces[0].Namespace, dependencyEdges, dependentEdges)
+
+				// Update the edges with pkgId and depPkgId.
+				dependentEdges[depPkgId] = append(dependentEdges[depPkgId], pkgId) // pkgId is dependent on depPkgId
+				dependencyEdges[pkgId] = append(dependencyEdges[pkgId], depPkgId)  // depPkgId is a dependency of pkgId
+			}
+		}
+	}
+
+	return packages, nil
+}
+
+// updatePackagesAndNames updates the mapping of package IDs to their names, and constructs the dependency graph.
+// It takes a set of parameters including maps for ID to name conversion, packages, dependency and dependent edges,
+// and information about the package and its dependency such as their IDs, names, types, and namespaces.
+// This function skips processing for "guac" files in the "files" namespace and updates the provided maps with
+// the relationships between packages and their dependencies. It leverages traverseGraph to find all packages
+// that are either dependencies of or dependents on the given package, and updates the packages map accordingly.
+func updatePackagesAndNames(idToName map[string]string, packages map[string]dependencyNode, depPkgId, pkgId, depPkgName, pkgName, depPkgType, depPkgNamespace string, dependencyEdges, dependentEdges map[string][]string) {
+	// Skip "guac" files.
+	if depPkgType == "guac" && depPkgNamespace == "files" {
+		return
+	}
+
+	// Map the IDs to their names.
+	idToName[depPkgId] = depPkgName
+	idToName[pkgId] = pkgName
+
+	// First, we need to find all the packages that have pkgName as a dependency.
+	// Note that we are only searching of packages with pkgName as a dependency from the packages that have scanned so far.
+
+	// This dependencyPackages map finds all packages that have pkgName as a dependent out of our pre-scanned packages.
+	dependencyPackages := traverseGraph(depPkgId, dependencyEdges)
+
+	// Next we want to find all the packages that are dependencies of pkgName.
```

Comment:
What follows this comment finds the *dependents* of `pkgName`, or alternatively the packages that *have* `pkgName` as a dependency.  Also see my other review comment -- I think that at some point the order of the `traverseGraph` calls was swapped but these comments did not go with it? 

Also, since this wording can be tricky, I would recommend sticking to either the direct form of `the packages that are dependencies / dependents of foo` or the indirect form of `the packages that have foo as a dependent / dependency`,  instead of swapping between the two.

Reply:
Done. The comments were getting kind of confusing before, thanks for the reviews!